### ISSUE TITLE
mape metric not working in gamma regression with .cv function

### ISSUE THREAD

**Issue by rehan814** [microsoft/LightGBM#2878]
If I set metric for evaluation as 'mape' with objective set to 'gamma', the output of the .cv function is the same as if I set metric to 'l1'. 

see the output  of cv function below. 

pd.DataFrame(cv_output)
Out[2]: 
    mape-mean  mape-stdv   l1-mean   l1-stdv   l2-mean   l2-stdv
0    0.003272   0.000064  0.003272  0.000064  0.000224  0.000048
1    0.003007   0.000067  0.003007  0.000067  0.000222  0.000048
2    0.002864   0.000068  0.002864  0.000068  0.000222  0.000048
3    0.002739   0.000067  0.002739  0.000067  0.000221  0.000048
4    0.002637   0.000067  0.002637  0.000067  0.000220  0.000048
5    0.002561   0.000067  0.002561  0.000067  0.000220  0.000048

Is this a design choice for gamma regression or is it a bug?

my parameters are as follows. 

 params_model:
 objective: 'gamma'
 boosting_type: 'gbdt'
 learning_rate: 0.05
 nthread: 4
 verbosity: -12

 params_cv:
 'seed': 10
 'num_boost_round': 500
 'verbose_eval': False
 'metrics': ['mape', 'l1', 'l2']
 'early_stopping_rounds': 100

## Environment info

Operating System:
Linux 14.04.1-Ubuntu SMP Thu May 9 08:17:37 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

CPU/GPU model:
CPU:       Hexa core Intel Xeon W-2133 CPU (-HT-MCP-) cache: 8448 KB flags: (lm nx sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx)
           Clock Speeds: 1: 3799.968 MHz 2: 3799.968 MHz 3: 3799.968 MHz 4: 3799.968 MHz 5: 3799.968 MHz 6: 3799.968 MHz 7: 3799.968 MHz 8: 3799.968 MHz 9: 3799.968 MHz 10: 3799.968 MHz 11: 3799.968 MHz 12: 3799.968 MHz

C++/Python/R version:
Python 3.7

LightGBM version or commit hash:
2.3.1

## Reproducible examples
I tried to reproduce the similar behavior using boston dataset from sklearn but apparently mape and l1 values are different for this dataset.     

mape-mean  mape-stdv   l1-mean   l1-stdv    l2-mean   l2-stdv
0    0.350261   0.018872  6.476793  0.293785  80.414499  8.142670
1    0.331261   0.017205  6.197715  0.309359  75.012017  8.139730
2    0.312782   0.016992  5.934062  0.316463  69.953947  7.936918
3    0.301183   0.016775  5.754473  0.308817  66.491511  7.425545
4    0.286844   0.016526  5.532690  0.312193  62.582373  7.179733

So, I assume there is some condition somewhere in the code that gets triggered and mape metric gets changed to l1. 

If thats the case then what and where is this condition. 

I also looked at the predictions of my dataset after 1, 10 iterations and there is no NaNs if I calculate Mape on them.

**Comment by guolinke** [comment]
refer to https://github.com/microsoft/LightGBM/blob/9654f16528d2d1e3462a298106cb1fc6ed43cbac/src/metric/regression_metric.hpp#L249

I guess your labels are not greater the 1.
